docs/en/notes/contribution_guide.md

```diff
@@ -1,31 +1,234 @@
-- If you plan to add some new features that involve large changes, it is encouraged to open an issue for discussion first.
-- If you are the author of papers in this field and would like to include your work to MMPose, please contact us. We will much appreciate your contribution.
+Here’s a brief introduction to the origin and upstream. When we use “git clone”, we create an “origin” remote by default, which points to the repository cloned from. As for “upstream”, we add it ourselves to point to the target repository. Of course, if you don’t like the name “upstream”, you could name it as you wish. Usually, we’ll push the code to “origin”. If the pushed code conflicts with the latest code in official(“upstream”), we should pull the latest code from upstream to resolve the conflicts, and then push to “origin” again. The posted Pull Request will be updated automatically.
 ```
 
-## Code style
+### 2. Configure pre-commit
 
-### Python
+You should configure pre-commit in the local development environment to make sure the code style matches that of OpenMMLab. Note: The following code should be executed under the MMPOSE directory.
+
+```Shell
+pip install -U pre-commit
+pre-commit install
+```
+
+Check that pre-commit is configured successfully, and install the hooks defined in `.pre-commit-config.yaml`.
+
+```Shell
+pre-commit run --all-files
+```
+
+![](https://user-images.githubusercontent.com/57566630/202368856-0465a90d-8fce-4345-918e-67b8b9c82614.png)
+
+```{note}
+Chinese users may fail to download the pre-commit hooks due to the network issue. In this case, you could download these hooks from:
+
+pip install -U pre-commit -i https://pypi.tuna.tsinghua.edu.cn/simple
+
+or:
+
+pip install -U pre-commit -i https://pypi.mirrors.ustc.edu.cn/simple
+```
+
+If the installation process is interrupted, you can repeatedly run `pre-commit run ...` to continue the installation.
+
+If the code does not conform to the code style specification, pre-commit will raise a warning and fixes some of the errors automatically.
+
+![](https://user-images.githubusercontent.com/57566630/202369176-67642454-0025-4023-a095-263529107aa3.png)
+
+### 3. Create a development branch
+
+After configuring the pre-commit, we should create a branch based on the dev branch to develop the new feature or fix the bug. The proposed branch name is `username/pr_name`.
+
+```Shell
+git checkout -b username/refactor_contributing_doc
+```
 
-We adopt [PEP8](https://www.python.org/dev/peps/pep-0008/) as the preferred code style.
+In subsequent development, if the dev branch of the local repository lags behind the dev branch of the official repository, you need to pull the upstream dev branch first and then rebase it to the local development branch.
 
-We use the following tools for linting and formatting:
+```Shell
+git checkout dev-1.x
+
+git pull upstream dev-1.x
+
+git checkout username/refactor_contributing_doc
+
+git rebase dev-1.x
```

Comment:
By default, the local dev-1.x will track origin/dev-1.x instead of upstream/dev-1.x, so should it be like:

```shell
git checkout username/refactor_contributing_doc
git fetch upstream
git rebase upstream/dev-1.x
```